### DISCUSSION TITLE
Documentations on render repeater

### DISCUSSION THREAD

**Discussion by MaartenJC** [area17/twill#1201] [Q&A]
I had a hard time to find out how to render a repeater in a block.

This is the solution i found out after searching several posts, but couldn't find it in the docs. Is this the right way to do it? Would it be an option to explain it better in the docs?

Block select.blade.php:
```
@formField('repeater', ['type' => 'select_item'])
```
Repeaters: select_item.blade.php:
```
@twillRepeaterTitle('Select item')
@twillRepeaterMax('10')

@formField('input', [
    'name' => 'label',
    'type' => 'text',
    'label' => 'Label'
])

@formField('input', [
    'name' => 'value',
    'type' => 'text',
    'label' => 'Value'
])
```
Block rendering:
```
<select name="{!! $block->input('name') !!}">
  @foreach ($block->children as $child)
    <option value="{{ $child->input('name') }}">{{ $child->input('label') }}</option>
  @endforeach
</select>
```

**Comment by pboivin** [comment]
Hey @mcouvreur, I agree that the relationship between blocks/repeaters, and perhaps the overall workflow is underdocumented. Feel free to suggest any improvements.

So your solution looks great if you want really granular control of how the repeater items are rendered. You can also define frontend views for your repeater items, just like blocks. Quick example —

For context: 

- A `pages` module has a `block_editor` field
- There is a block called `customlist`
- This block has a repeater for `customlist-item`

File structure:
```
resources/
    views/
        admin/
            blocks/
                - customlist.blade.php          | Form templates
            repeaters/                          |                       
                - customlist-item.blade.php     |
        site/
            blocks/
                - customlist.blade.php          | Frontend templates
                - customlist-item.blade.php     |
```

On the frontent, this would render the `customlist` template, followed by all children `customlist-item`:
```
{!! $page->renderBlocks() !!}
```

The first parameter of `renderBlocks` can be used to disable rendering the child blocks:
```
{!! $page->renderBlocks(false) !!}
```
And this gives you full control for rendering, like you do in your example.

Hope this can help!


https://twill.io/docs/api/2.x/A17/Twill/Models/Behaviors/HasBlocks.html
